### PR TITLE
Add Viper parametrization as E2E config option.

### LINKED ISSUE #18099: Proposal for e2e configuration parameters.
**Problem Statement:**

Currently the e2e tests have too many hard coded and command line parameters for testing options. 

**Proposal:**

Centralize the configuration of the parameters into a single configuration object that can be easily: 

- serialized (json) 
- auto-generated with sane defaults
- user editable 
- hierarchical based on test
```
e2e: {
    gloablparam1 : "foo"
    testA: { 
          test_param1
    }
    density: { 
          max_pods_per_node
    }
}
```

/cc @kubernetes/sig-scalability @wojtek-t  @ixdy @jayunit100 @gmarek 
xref: #18097 , subsumes: #10479

### PR DESCRIPTION
**Pull request by jayunit100**
do-not-merge 

Fixes #18099 via viper rather than inis.  
Wont build until we remove BurntSushi/ COPYING based deps from upstream viper.
I'll dig into those issues independently and update later, before pushing the updated godeps into this PR.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31030)

### CODE REVIEW

**Comment by jayunit100** [comment]
not sure why it beaks node e2e.  for now will wait for other results to come in.

**Comment by jayunit100** [comment]
Unknown flag test.timeout

**Comment by jayunit100** [comment]
Probably another flag set I need to integrate to pflag/viper will dig....

**Comment by ixdy** [comment]
Can you put the vendor import in a separate commit please? This is pretty hard to review otherwise.

**Comment by jayunit100** [comment]
i think this is ready for a look now.  cleaned up and minimized.  gave up on deep flag integration, added a comment as to why :)

File: test/e2e/framework/test_context.go [hunk 1]
```diff
@@ -172,3 +175,29 @@ func RegisterNodeFlags() {
...
+	// TODO @jayunit100: Maybe a more elegant viper-flag integration for the future?
+	// For now, we layer it on top, because 'flag' deps of 'go test' make pflag wrappers
+	// fragile, seeming to force 'flag' to have deep awareness of pflag params.
+	RegisterCommonFlags()
+	RegisterClusterFlags()
+	flag.Parse()
+
+	viperFlagSetter := func(f *flag.Flag) {
+		if viper.IsSet(f.Name) {
+			fmt.Println("[overwrite] viper found a Settting for  ", f.Name, " ", f.Value)
```
Comment (timothysc) [inline test/e2e/framework/test_context.go]: s/Settting/Setting

File: test/e2e/framework/test_context.go [hunk 2]
```diff
@@ -34,7 +36,8 @@ type TestContextType struct {
 	CertDir            string
 	Host               string
 	// TODO: Deprecating this over time... instead just use gobindata_util.go , see #23987.
-	RepoRoot       string
+	RepoRoot string
```
Comment (timothysc) [inline test/e2e/framework/test_context.go]: probably don't need this change here.

File: test/e2e/framework/test_context.go (hunk 1 above)
Comment (timothysc) [inline test/e2e/framework/test_context.go]: Do we want to print this all out, seems more like debug only level printing.   We have so much spam today, and I think only the sig-testing folks care about this.

Comment (jayunit100) [inline test/e2e/framework/test_context.go]: ill replace w/ glog.V(4)

**Comment by jayunit100** [comment]
updated logging etc.  the `RepoRoot` tab change is induced via gofmt, but its a non-essential change (either way is gofmt compliant)

File: test/e2e/framework/test_context.go (hunk 2 above)
Comment (jayunit100) [inline test/e2e/framework/test_context.go]: Gofmt keeps adding this back for me.  I think it's the default?

**Comment by timothysc** [comment]
I don't really see any problem with this change but we are in 1.4 code freeze so I'd like someone from google to weigh in. 

/cc @kubernetes/sig-testing @ixdy @spxtr

**Comment by ixdy** [comment]
yeah, I'm inclined to say this waits for 1.5. :(

@pwittrock

**Comment by ixdy** [comment]
per https://groups.google.com/d/msg/kubernetes-dev/pdMJpA0NF7I/1dXrRBriCQAJ, it seems likely this will slip to 1.5. I intend to review it still, though.

**Comment by pwittrock** [comment]
1.5 SGTM

**Pull request merged**